modules/trusted-firmware-m/CMakeLists.txt

```diff
@@ -35,6 +35,18 @@ if (CONFIG_BUILD_WITH_TFM)
     list(APPEND TFM_CMAKE_ARGS -DBL2=TRUE)
     list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_IMAGE_VERSION_S=${CONFIG_TFM_IMAGE_VERSION_S})
     list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_IMAGE_VERSION_NS=${CONFIG_TFM_IMAGE_VERSION_NS})
+
+    list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_SIGNATURE_TYPE=${CONFIG_TFM_MCUBOOT_SIGNATURE_TYPE})
+    if ("${CONFIG_TFM_MCUBOOT_SIGNATURE_TYPE}" MATCHES "^RSA")
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_USE_PSA_CRYPTO=OFF)
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_HW_KEY=ON)
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_BUILTIN_KEY=OFF)
+    elseif ("${CONFIG_TFM_MCUBOOT_SIGNATURE_TYPE}" MATCHES "^EC-P")
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_USE_PSA_CRYPTO=ON)
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_HW_KEY=OFF)
+      list(APPEND TFM_CMAKE_ARGS -DMCUBOOT_BUILTIN_KEY=ON)
```

Comment:
Why these options based on the type of key ? the only dependency that is clear to me is `MCUBOOT_BUILTIN_KEY` requires `MCUBOOT_USE_PSA_CRYPTO` but I don't understand why, for example, you set HW_KEY for RSA and BUILTIN_KEY for EC-P.

Maybe just a comment explaining it would make things easier.

Reply:
I added a comment to explain why.
RSA without `MCUBOOT_HW_KEY` and EC-P without `MCUBOOT_BUILTIN_KEY` failed to boot (`Image in the primary slot is not valid!` and `Provisioning failed`, respectively).